timeseries/time_series.go

```diff
@@ -35,6 +35,38 @@ func (ts *TimeSeries) AddCandle(c *Candle) error {
 	return ErrUnexpectedTime
 }
 
+// RemoveCandles remove the candle from series
+// [startIndex] is mandatory
+// [endIndex] is optional
+// returns a new timeseries object
+func (ts *TimeSeries) RemoveCandles(startIndex int, endIndex *int) (*TimeSeries, error) {
```

Comment:
I think it would be better to pass `endIndex` as `int` (not pointer). If pass 0 then `endIndex` is not set. It seems that pointer is not needed here

Initially we can assign length of time series if `endIndex` equals zero. Further `endIndex` checks can be removed
```
if endIndex == 0 {
   endIndex = ts.Length()
}
```

Reply:
I assign the `endIndex` as `pointer` to make it **optional** parameter.

What do you think about doing `variadic function parameter` for `endIndex`?

Reply:
I think `variadic function parameter` complicates the function signature. It seems to use "default value as no value" is normal for golang

Reply:
@evsamsonov I made the changes, you can check it. thanks for the tips & tricks.

Reply:
@kaankoken Thanks!